pkgs/build-support/cc-wrapper/add-hardening

```diff
@@ -0,0 +1,61 @@
+hardeningFlags=(fortify stackprotector pic strictoverflow format relro bindnow)
+hardeningFlags+=("${hardeningEnable[@]}")
+hardeningCFlags=()
+hardeningLDFlags=()
+hardeningDisable=${hardeningDisable:-""}
+
+if [[ "$($LD -z 2>&1)" =~ "unknown option" ]]; then
```

Comment:
Doesn't this cause an extra invocation of `ld` for every call to the gcc wrapper? That seems pretty expensive to me...